ff/serving/provider/offline.go

```diff
@@ -89,13 +111,81 @@ func (store *memoryOfflineStore) CreateResourceTable(id ResourceID) (OfflineTabl
 }
 
 func (store *memoryOfflineStore) GetResourceTable(id ResourceID) (OfflineTable, error) {
+	return store.getMemoryResourceTable(id)
+}
+
+func (store *memoryOfflineStore) getMemoryResourceTable(id ResourceID) (*memoryOfflineTable, error) {
 	table, has := store.tables[id]
 	if !has {
 		return nil, &TableNotFound{id.Name, id.Variant}
 	}
 	return table, nil
 }
 
+// Used to implement sort.Interface for sorting.
+type materializedRecords []ResourceRecord
+
+func (recs materializedRecords) Len() int {
+	return len(recs)
+}
+
+func (recs materializedRecords) Less(i, j int) bool {
+	return recs[i].Entity < recs[j].Entity
+}
+
+func (recs materializedRecords) Swap(i, j int) {
+	recs[i], recs[j] = recs[j], recs[i]
+}
+
+func (store *memoryOfflineStore) CreateMaterialization(id ResourceID) (Materialization, error) {
+	if id.Type != Feature {
+		return nil, errors.New("Only features can be materialized")
+	}
+	table, err := store.getMemoryResourceTable(id)
+	if err != nil {
+		return nil, err
+	}
+	matData := make(materializedRecords, 0, len(table.entityMap))
+	for _, records := range table.entityMap {
+		matRec := latestRecord(records)
+		matData = append(matData, matRec)
+	}
+	sort.Sort(matData)
+	matId := MaterializationID(uuid.NewString())
+	mat := &memoryMaterialization{
+		id:   matId,
+		data: matData,
+	}
+	store.materializations[matId] = mat
+	return mat, nil
+}
+
+type MaterializationNotFound struct {
+	id MaterializationID
+}
+
+func (err *MaterializationNotFound) Error() string {
+	return fmt.Sprintf("Materialization %s not found", err.id)
+}
+
+func (store *memoryOfflineStore) GetMaterialization(id MaterializationID) (Materialization, error) {
```

Comment:
Should the materialization id somehow be tied to the resource id of the table that generates it, so a store's tables and materializations can be connected via some built-in metadata definition?